# 2012 CK positions mapped to 2018 human reference positions

8.26.19
- original markdown of project made 8.12.19

mfpfox

---

###### track version hisotry with 
> ‘entry name tracker’ tool 


**VOCAB:**
* **primary (citable) accession number** = stable identifiers used to cite UKB entries. each entry is assigned a unique accession number. 
    more info below <https://www.uniprot.org/help/accession_numbers>

* **protein isoform name** = 
    - Given that for most genes there is no accepted, standardized nomenclature that covers all individual isoforms, our policy is to identify them using simple numbering, beginning with isoform 1. Example: Q15406
    - When a comprehensive nomenclature does exist for all isoforms, then this will be applied. Example: P04150
    - In either case all isoform-specific synonyms used in the literature will also be recorded, in order to facilitate the identification of specific isoform sequences by users. Example: P08235


* **UKB stable ID == primary accession for entry**

---

## are all uniprot proteins unique??
###### only detected 3979 duplicated protein sequence:
**P86791 & P86790**

<https://uswest.ensembl.org/Homo_sapiens/Transcript/ProteinSummary?db=core;g=ENSG00000122674;r=7:5898733-5926550;t=ENST00000325974>
> P86791 (CCZ1_HUMAN) ENST00000325974; ENSP00000325681; ENSG00000122674
    
    > CCZ1 homolog, vacuolar protein trafficking and biogenesis associated [Source:HGNC Symbol;Acc:HGNC:21691]

https://uswest.ensembl.org/Homo_sapiens/Transcript/ProteinSummary?db=core;g=ENSG00000146574;r=7:6794134-6826770;t=ENST00000316731>
> P86790 (CCZ1B_HUMAN) ENST00000316731; ENSP00000314544; ENSG00000146574

    > CCZ1 homolog B, vacuolar protein trafficking and biogenesis associated [Source:HGNC Symbol;Acc:HGNC:21717]

---
---
---

# PROBLEM (before keri shared og file)

## from Keri's Cys paper: database (release-2012_11)

### only available release from 2012 is 01_2012, see UKB README from FTP site below...
<ftp://ftp.uniprot.org/pub/databases/uniprot/previous_releases/>

### also testing 01_2013 which is less time between actual data release vs 01_2012
<ftp://ftp.uniprot.org/pub/databases/uniprot/previous_releases/release-2013_01/knowledgebase/>


Universal Protein Resource (UniProt)
====================================

The Universal Protein Resource (UniProt), a collaboration between the European
Bioinformatics Institute (EBI), the SIB Swiss Institute of Bioinformatics, and
the Protein Information Resource (PIR), is comprised of three databases, each
optimized for different uses. The UniProt Knowledgebase (UniProtKB) is the
central access point for extensively curated protein information, including
function, classification and cross-references. The UniProt Reference Clusters
(UniRef) combine closely related sequences into a single record to speed up
sequence similarity searches. The UniProt Archive (UniParc) is a comprehensive
repository of all protein sequences, consisting only of unique identifiers and
sequences.

The UniProt Knowledgebase (UniProtKB) has been created from Swiss-Prot, TrEMBL
and PIR-PSD. It consists of two parts, one containing fully manually annotated
records and another one with computationally analysed records awaiting full
manual annotation. The two sections will be referred to as the Swiss-Prot
section of the UniProt Knowledgebase (UniProtKB/Swiss-Prot) and TrEMBL section
of the UniProt Knowledgebase (UniProtKB/TrEMBL), respectively. PIR-PSD release
80.0 of 31-Dec-2004 has been fully integrated into these sections. This was the
last release of PIR-PSD.

This directory, 
/uniprot/previous_releases/, 
contains previous releases of the UniProt Knowledgebase
and of UniRef.
**Archives of each release are kept for 2 years.
For older releases only major and first releases of each year are kept.**

The directory itself contains two subdirectories:

subdirectory /knowledgebase:

contains three compressed tar files,
knowledgebaseYYYY_AA.tar.gz archiving 
   uniprot_sprot.dat, 
   uniprot_trembl.dat
   and the /docs directory 
uniprot_sprot-onlyYYYY_AA.tar.gz archiving
   uniprot_sprot.dat,
   uniprot_sprot.fasta,
   uniprot_sprot_varsplic.fasta
   and uniprot_sprot.xml
knowledgebase-docs-onlyYYYY_AA.tar.gz (from release 2019_02) archiving
   the /docs directory
and the files,
  UniProtKB_SwissProt-relstat.html (from release 2011_01)
  UniProtKB_TrEMBL-relstat.html (from release 2011_01)
for releaseYYYY_AA 

subdirectory /uniref:

contains one compressed tar file, unirefYYYY_AA.tar.gz,
archiving 3 separate files
   uniref50.xml,
   uniref90.xml 
   uniref100.xml 
for releaseYYYY_AA

Note:
The current release of the UniProt Knowledgebase and UniRef is 
available at /uniprot/current_release (in flat file, XML and FASTA formats,
as well as by taxonomic division).


## 2012_01 & 2013_01 processing [fasta to csv]

downloaded 11.5.19, shell notes 
```bash
python /Users/mariapalafox/Desktop/Toolbox/processUniprotFasta.py uniprot_sprot.fasta uniprot_sprot_fasta.csv uniprot

grep '_HUMAN' uniprot_sprot_fasta.csv > human_uniprot_sprot.csv

# 2012
wc -l human_uniprot_sprot.csv  # 20242 human_uniprot_sprot.csv (no header)

# 2013
wc -l human_uniprot_sprot.csv # 20232 human_uniprot_sprot.csv (no header)

# added header
cat header.csv human_uniprot_sprot.csv > human_sprot_2012_01.csv or 2013 version saved

```

## comparing to set of 4119 IDs from merged CK detected subset data:
```python
# importing 2012 fasta canonical seq
ukb12 = pd.read_csv("human_sprot_2012_01.csv")
# importing 2013 fasta canonical seq
ukb13 = pd.read_csv("human_sprot_2013_01.csv")
```
### results of ID searching:
#### 01_2013 has 4088/4119 (-31) 
#### 01_2012 has 4090/4119 (-29)
#### 2018 has 4000/4119 (-119) 

---
---
---
---
---
---

# ABOUT Cravatt lab original fasta file
[uniprot fasta format](https://www.uniprot.org/help/fasta-headers)

- included '>tr' rows, for UniProtKB/TrEMBL
    - some ID may not be found because we are only searching in 2018 SWISSPROT CCDS set
 
| database |ID count |
|----------|-------|
| swissprot| 18832 |
| trembl   | 1149  |
| total    | 19981 |
| tot. detected | 4119|
| tot. in 11_2012 | 4093|

## [idea!]

**grab the SV=# information from 1st line of entry, which will help analysis to explain why seqeunces are different, "Q12345 pos not found because teh SV was 1 in 2012 and 10 in 2018**

**grab the PE=# info for protein existence level information**

# accounting in MAPPING_tables.xlsx
- directory of project: "/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/"

## DETECTED subsets - input files: 
1. LYS_reactivity_2012_4579.csv
2. LYS_pan_target_2012_8360.csv
3. CYS_reactivity_2012_1541.csv
4. CYS_pan_target_2012_6157.csv

### 2018 total C and K in canonical sequence file:
**Kpos_CCDSset.csv** == 614,333 lysines in 2018 CCDS UKB

**Cpos_CCDSset.csv** == 242,940 cysteines in 2018 CCDS UKB

---

# REASON why some Nature positions dont match CCDS CYS LYS positions: 
1. entry ID has been updated
2. entry has been merged with another 
3. sequence corresponding to entry has changed so CYS no longer in same pos

**biostar related comment:** "The reason why these numbers do not correspond to the numbers from the release notes is that there have been deletions, merges and demerges, and entries that were present in 2014_09 may no longer exist now."

---
---
---

In [2]:
import os
import sys
import pandas as pd
import numpy as np

In [51]:
sys.path.append("/Users/mariapalafox/Desktop/Toolbox")
from all_funx import *

In [4]:
from IPython.display import display
pd.set_option('display.max_columns', None)
pd.options.display.max_seq_items = 2000

from IPython.core.display import display, HTML
display(HTML("<style>.container {width:90% !important;}</style>"))

In [5]:
os.chdir("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/")
print(os.listdir())

['CCDS2018_all_Cpos.csv', 'human_sprot_2012_01.csv', 'CYS_pan_target_2012_6157.csv', '.DS_Store', 'CODEnotes_mapCK_v0_attempt.sh', 'LYS_reactivity_2012_4579.csv', 'UniProt_Human_Cravattlab_nonredundant2_98id_11_05_2012_DELreversed.fasta', 'CYS_reactivity_2012_1541.csv', 'MAPPED_CK_v1', 'NATURE2012_UKBccds2018_files', 'CCDS2018_all_Kpos.csv', 'LYS_pan_target_2012_8360.csv', '2012_01_UKB_Swiss_release', 'human_sprot_2013_01.csv', 'cravattlab_sp_reviewed_ID_only_18832.txt', 'cravattlab_tr_unrevied_ID_only_1149.txt', 'Reprocesssing2012_Nature_allCYSLYSdata.xlsx', 'UKB_CCDS_2018_06_fasta.csv', '2013_01_UKB_Swiss_release', 'CoordinatesP06733_K64.txt', 'cravattlab_11_2012.csv', 'UniqueIDs_og_data_2012_11_uniprot_sprot_4119.csv']


# [0] how many uniprot IDs total were detected in chemoproteomic experiments?

In [53]:
# importing 2018 fasta canonical seq
ukb18 = pd.read_csv("UKB_CCDS_2018_06_fasta.csv")

# importing 2018 CK positions
#Cysnature = pd.read_csv("CCDS2018_all_Cpos.csv", header=None)
#Lysnature = pd.read_csv("CCDS2018_all_Kpos.csv", header=None)

# importing 2012 database searched to generate CK positions
chempro = pd.read_csv("cravattlab_11_2012.csv")

# importing cys react and pan/target
Creact = pd.read_csv("CYS_reactivity_2012_1541.csv",header=None)
Cpan = pd.read_csv("CYS_pan_target_2012_6157.csv",header=None)

# importing lys react and pan/target
Kreact = pd.read_csv("LYS_reactivity_2012_4579.csv",header=None)
Kpan = pd.read_csv("LYS_pan_target_2012_8360.csv",header=None)

**merge all chemoproteomic data IDs into set of all unique IDs ever detected "origianl ID count"**

In [7]:
idCR = Creact[[0]] 
idCP = Cpan[[0]]
idKR = Kreact[[0]] 
idKP = Kpan[[0]]
mer2012 = pd.concat([idCR, idCP, idKR, idKP])
mer2012.head(3)

0
0   Q9Y6W6_C177
1    Q9Y6M9_C42
2  Q9Y6J0_C1197

In [8]:
def splitID(df1, col):
    new1 = df1[col].str.split("_", n=1, expand = True)
    df1["ID"] = new1[0]
    # return set of IDs
    setID = set(df1.ID)
    return setID

In [10]:
# split merged chemo data to isolate all unique IDs
IDsetog = splitID(mer2012, 0)
print(len(IDsetog))

4119


In [11]:
IDog = list(IDsetog)
IDog = pd.DataFrame(IDog)

In [12]:
IDog.columns = ['unique_IDs']
IDog.head(5)

unique_IDs
0     Q8IZ69
1     Q9UHQ1
2     Q96RL7
3     Q9UL12
4     Q9NRL3

In [13]:
# saving this file to label 2012 and 2018 fasta-csv files for detected ID match
IDog.to_csv("UniqueIDs_og_data_2012_11_uniprot_sprot_4119.csv", index=False)

In [14]:
# 2018 FASTA
dropUnnamed(ukb18)
ukb18.head(3)

ID    entryName  Length  \
0  O95139  NDUB6_HUMAN     128   
1  O75438  NDUB1_HUMAN      58   
2  Q8N4C6    NIN_HUMAN    2090   

                                         proSequence  
0  MTGYTPDEKLRLQQLRELRRRWLKDQELSPREPVLPPQKMGPMEKF...  
1  MVNLLQIVRDHWVHVLVPMGFVIGCYLDRKSDERLTAFRNKSMLFK...  
2  MDEVEQDQHEARLKELFDSFDTTGTGSLGQEELTDLCHMLSLEEVA...

In [15]:
chempro.head(4)

ID    entryName  Length  \
0  Q8NGR1  O13A1_HUMAN     328   
1  Q8N6Q8  MET25_HUMAN     603   
2  Q9UBK2  PRGC1_HUMAN     798   
3  P35998   PRS7_HUMAN     433   

                                         proSequence  
0  MKLWMESHLIVPETRPSPRMMSNQTLVTEFILQGFSEHPEYRVFLF...  
1  MAASCPLPVTPDLPTLRAKLQGLLQFLRDALSISNAHTVDFYTESV...  
2  MAWDMCNQDSESVWSDIECAALVGEDQPLCPDLPELDLSELDVNDL...  
3  MPDYLGADQRKTKEDEKDDKPIRALDEGDIALLKTYGQSTYSRQIK...

In [16]:
# ADDING COLUMN AND DROPPING ROWS WITH NO ID MATCH
IDlistog = IDog.unique_IDs.tolist()
db18 = addcolumnconditionalDrop(IDlistog, ukb18, 'ID', 'detected_2012_ID') 
dbchem = addcolumnconditionalDrop(IDlistog, chempro, 'ID', 'detected_2012_ID') 

**variables: db18- database 2018 with only rows with 4119 IDs, dbchem- MS database where all IDs should be found and all positions found**

In [17]:
print(len(IDlistog)) # 4119 original IDs in dataset

4119


In [47]:
describeMe(dbchem)
describeMe(db18)

(4093, 5)
Index(['ID', 'entryName', 'Length', 'proSequence', 'detected_2012_ID'], dtype='object')
ID                  0
entryName           0
Length              0
proSequence         0
detected_2012_ID    0
dtype: int64
(4000, 5)
Index(['ID', 'entryName', 'Length', 'proSequence', 'detected_2012_ID'], dtype='object')
ID                  0
entryName           0
Length              0
proSequence         0
detected_2012_ID    0
dtype: int64


## which IDs from merged chemoproteomic data were not found in MS database file and 2018 database?

In [20]:
# IDs in 2012 data but not in 01_2012 uniprot fasta
det12_list = list(dbchem.ID)
det18_list = list(db18.ID)

notin12 = diffList(IDlistog, det12_list)
print(notin12)
print(len(notin12))
print()


notin18 = diffList(IDlistog, det18_list)
print()
print(notin18)
print(len(notin18))

['Q5VTE0', 'P83881', 'P68363', 'B2RPK0', 'A6NEC2', 'Q8NHW5', 'A6NNZ2', 'P04350', 'Q6P474', 'P86791', 'P26640', 'Q9BTY7', 'Q99613', 'A6NKZ1', 'Q9BQE3', 'Q5VYJ4', 'P63241', 'P58107', 'P19105', 'Q9GZM3', 'P0C860', 'P63261', 'Q71DI3', 'A6NF79', 'P0DMV9', 'Q8IYB9']
26


['P62861', 'Q9ULA0', 'Q5VTE0', 'P83881', 'P07988', 'Q9Y5J9', 'J3QR44', 'P0CB43', 'E7ETI0', 'I3L499', 'Q6P474', 'Q96I15', 'O94854', 'A6NKZ1', 'Q8WX92', 'H0YGG7', 'P49411', 'P30460', 'P61758', 'O43847', 'P62158', 'F5H5P2', 'Q8N5W9', 'H3BN57', 'Q5VYK3', 'P05534', 'P56556', 'P0CG05', 'Q8IXH7', 'C9J4G0', 'O00422', 'Q7Z3J2', 'Q86YG4', 'Q8WYJ6', 'H0Y2S0', 'E7EQZ4', 'I3L3Q1', 'A6NEC2', 'Q8NHW5', 'Q8NAG6', 'A6NNZ2', 'Q8IY67', 'Q9NYZ3', 'Q9Y6A9', 'Q9NVE7', 'Q6DKI1', 'A6NE09', 'Q9Y618', 'P29034', 'A6NF79', 'Q7Z460', 'P42858', 'P22061', 'E7EVK2', 'H3BQZ7', 'Q9H3P2', 'Q5VTU8', 'Q8N9T8', 'Q8NDA2', 'P57053', 'B2RPK0', 'P0CW22', 'O15027', 'Q53H96', 'Q5JZ73', 'P17405', 'Q96EY9', 'Q9H3K6', 'Q9Y4D8', 'P0DI82', 'P01892', 'P01871', 'D6RAD4', 'Q9

### checking the IDs that were in the chemoproteo. CK set but not in the MS database from 11_2012...

* Caution, could be product of pseudogene: Q5VTE0;B2RPK0;Q8NHW5;Q6P474;P58107 (uncertain of MET initiator);P0C860;Q71DI3

* obsolete: A6NF79;
 
* ribosomal protein: P83881;

* homolog: Q9BTY7 (HGH1 homo); 

* **should be in MS database: P68363 (tubulin);A6NEC2 (aminopeptidase);A6NNZ2 (tubulin);P04350 (tubulin);P86791 (guanine exchange factor); P26640 (tRNA ligase);P63241 (mRNA binding protein);P19105 (myosin light chain);Q9GZM3 (DNA dep. RNA pol);P63261 (actin); P0DMV9 (heat shock); Q8IYB9 (zinc finger)**

In [24]:
dbchem.reset_index(inplace=True, drop=True)
dbchem.sample(10)

ID    entryName  Length  \
3417  O43809  CPSF5_HUMAN     227   
3279  P67936   TPM4_HUMAN     248   
2190  P48960   CD97_HUMAN     835   
344   Q9NPI6  DCP1A_HUMAN     582   
71    Q99259   DCE1_HUMAN     594   
1265  Q14240  IF4A2_HUMAN     407   
1909  Q9NS86  LANC2_HUMAN     450   
1230  P20042   IF2B_HUMAN     333   
1599  Q9NZZ3  CHMP5_HUMAN     219   
3049  Q9NS87  KIF15_HUMAN    1388   

                                            proSequence detected_2012_ID  
3417  MSVVPPNRSQTGWPRGVTQFGNKYIQQTKPLTLERTINLYPLTNYT...             True  
3279  MAGLNSLEAVKRKIQALQQQADEAEDRAQGLQRELDGERERREKAE...             True  
2190  MGGRVFLAFCVWLTLPGAETQDSRGCARWCPQNSSCVNATACRCNP...             True  
344   MEALSRAGQEMSLAALKQHDPYITSIADLTGQVALYTFCPKANQWE...             True  
71    MASSTPSSSATSSNAGADPNTTNLRPTTYDTWCGVAHGCTRKLGLK...             True  
1265  MSGGSADYNREHGGPEGMDPDGVIESNWNEIVDNFDDMNLKESLLR...             True  
1909  MGETMSKRLKLHLGGEAEMEERAFVNPFPDYEAAAGALLASGAAEE...             True  
1230  MSGDEMIFDPTMSKKKKKKKKPFMLDEEGDTQTEETQPSETKEVEP...             True  
1599  MNRLFGKAKPKAPPPSLTDCIGTVDSRAESIDKKISRLDAELVKYK...             True  
3049  MAPGCKTELRSVTNGQSNQPSNEGDAIKVFVRIRPPAERSGSADGE...             True

In [25]:
dbchem.to_csv("sprot_11_2012_detectedIDs_4093.csv", index=False)

# [1] Formatting chemoproteomic data to contain only 4093 IDs also in 11_2012 ukbsp file from cravatt lab

4119-26 = 4093

## [a] make sure subset files for cys and lys are non-redundant 
- (there is redundancy however between label_type files, like positions from panReactive set are also in LMH data)

In [54]:
# changing labels
labels = ["pos_ID","reactivity"]
Creact.columns = labels
Kreact.columns = labels
labelpan = ['pos_ID','labelType']
Kpan.columns=labelpan
Cpan.columns=labelpan

**pos counts added to MAPPING_tables.xlsx under 2012 chemoproteomic files**

In [55]:
# check for duplicates in group files
print(len(Creact.pos_ID))
print(len(set(Creact.pos_ID)))

print(len(Kreact.pos_ID))
print(len(set(Kreact.pos_ID)))

print(len(Cpan.pos_ID))
print(len(set(Cpan.pos_ID)))

print(len(Kpan.pos_ID))
print(len(set(Kpan.pos_ID)))

1541
1541
4579
4579
6157
6157
8360
8360


**NOTE:  Nature data does not have redundancy when you process the reactivity data seperate from the pan/target data**

In [56]:
# EVER LABELED 2012 CYS 
CYSever2012 = set(list(Creact.pos_ID) + list(Cpan.pos_ID))
print(len(CYSever2012))
# EVER LABELED 2012 LYS
LYSever2012 = set(list(Kreact.pos_ID) + list(Kpan.pos_ID))
print(len(LYSever2012))

6510
9327


In [57]:
print(1541 + 6157) # redundant CYS
print(6510 - 7698) # CYS overlap
print(4579+8360) # redundant LYS
print(12939 - 9327) # LYS overlap

7698
-1188
12939
3612


## [b] split pos_ID col and count # of unique positions in subsets and IDs

In [58]:
def split_ID(df1, col):
    new1 = df1[col].str.split("_", n=1, expand = True)
    df1["ID"] = new1[0]
    return df1

In [59]:
def count_unique_IDs(df, col):
    idset = set(df[col])
    print("unique IDs in df: ", len(idset))
    return idset

**ID counts added to MAPPING_tables.xlsx under 2012 chemoproteomic files**

In [60]:
# 2012 cys react group
crea = split_ID(Creact, 'pos_ID')
crea.head(3)

pos_ID    reactivity      ID
0   Q9Y6W6_C177  20.00 1/2/6   Q9Y6W6
1    Q9Y6M9_C42   5.74 1/1/6   Q9Y6M9
2  Q9Y6J0_C1197  20.00 1/2/6   Q9Y6J0

In [61]:
# 2012 cys pan group
cpant = split_ID(Cpan, 'pos_ID')
cpant.head(3)

pos_ID      labelType      ID
0  P26641_C266  Iodoacetamide  P26641
1  P07900_C374  Iodoacetamide  P07900
2  P23528_C139  Iodoacetamide  P23528

In [62]:
# 2012 lys react group
krea = split_ID(Kreact, 'pos_ID')
krea.head(3)

pos_ID  reactivity      ID
0  P52758_K117         0.1  P52758
1   P02766_K35         0.1  P02766
2   P51572_K79         0.2  P51572

In [63]:
# 2012 lys oan group
kpant = split_ID(Kpan, 'pos_ID')
kpant.head(3)

pos_ID labelType      ID
0  A0AVT1_K1014     IodoL  A0AVT1
1   A0AVT1_K544     IodoL  A0AVT1
2    A0AVT1_K86     IodoL  A0AVT1

## [c] label 4093 IDs chemoproteomic position data

In [65]:
det12_list = list(dbchem.ID)
print(len(det12_list))

4093


In [66]:
# ADDING COLUMN AND DROPPING ROWS WITH NO ID MATCH
Creact = addcolumnconditionalDrop(det12_list, crea , 'ID', '11_2012_detectedID') 
Cpan = addcolumnconditionalDrop(det12_list, cpant , 'ID', '11_2012_detectedID') 
Krea = addcolumnconditionalDrop(det12_list, krea , 'ID', '11_2012_detectedID') 
Kpant = addcolumnconditionalDrop(det12_list, kpant , 'ID', '11_2012_detectedID') 

In [68]:
print(Creact.shape)
print(Cpan.shape)
print(Krea.shape)
print(Kpant.shape)

(1483, 4)
(6157, 4)
(4576, 4)
(8350, 4)


In [74]:
Creact.to_csv("CYS_reactivity_11_2012IDs_1483.csv", index=False)
Creact.head(3)

pos_ID    reactivity      ID 11_2012_detectedID
0   Q9Y6W6_C177  20.00 1/2/6   Q9Y6W6               True
1    Q9Y6M9_C42   5.74 1/1/6   Q9Y6M9               True
2  Q9Y6J0_C1197  20.00 1/2/6   Q9Y6J0               True

In [75]:
Cpan.to_csv("CYS_pantarget_11_2012IDs_6157.csv", index=False)
Cpan.head(3)

pos_ID      labelType      ID 11_2012_detectedID
0  P26641_C266  Iodoacetamide  P26641               True
1  P07900_C374  Iodoacetamide  P07900               True
2  P23528_C139  Iodoacetamide  P23528               True

In [77]:
Krea.to_csv("LYS_reactivity_11_2012IDs_4576.csv", index=False)
Krea.head(3)

pos_ID  reactivity      ID 11_2012_detectedID
0  P52758_K117         0.1  P52758               True
1   P02766_K35         0.1  P02766               True
2   P51572_K79         0.2  P51572               True

In [78]:
Kpan.to_csv("LYS_pantarget_11_2012IDs_8350.csv", index=False)
Kpan.head(3)

pos_ID labelType      ID 11_2012_detectedID
0  A0AVT1_K1014     IodoL  A0AVT1               True
1   A0AVT1_K544     IodoL  A0AVT1               True
2    A0AVT1_K86     IodoL  A0AVT1               True

# doing 11_2012 pos checking against original chemoproteomic position data

```bash
~/Desktop/Toolbox/key_id_maker.py sprot_11_2012_detectedIDs_4093.csv sprot_11_2012_all_Cpos.csv C
#Number of amino acids in inputfile:  (44302, 1)
~/Desktop/Toolbox/key_id_maker.py sprot_11_2012_detectedIDs_4093.csv sprot_11_2012_all_Kpos.csv K
#Number of amino acids in inputfile:  (165363, 1)
```

### total C in 11_2012 filtered for 4093 IDs = 44,302
### total K in 11_2012 filtered for 4093 IDs = 165,363

In [79]:
def split_ID(df1, col):
    new1 = df1[col].str.split("_", n=1, expand = True)
    df1["ID"] = new1[0]
    return df1

def count_unique_IDs(df, col):
    idset = set(df[col])
    print("unique IDs in df: ", len(idset))
    return idset

# aalist is from fasta sequences, chemdf is chemoproteomic data subset
def position_overlap(aalist, chemdf, found_name, notfound_name):
    # create list of reactivity and pan target positions
    subsetls = list(chemdf.pos_ID)
    print("chemo pos_ID length: ", len(subsetls))
    # overlap with positions from fasta sequences
    overlapAA = set(aalist).intersection(subsetls)
    print("overlapping AA's in chemoproteomic data subset: ", len(overlapAA))
    # make dataframe with overlap results
    newls = list(overlapAA)
    newdf = pd.DataFrame(newls)
    newdf.columns = ['pos_ID']
    # count unique ID
    splitdf = split_ID(newdf, 'pos_ID')
    idset = count_unique_IDs(splitdf, 'ID')

    # label the subset data frame with new column whether its in fasta sequence or not
    done_overlapped = addcolumnconditional(overlapAA, chemdf, "pos_ID", "found_pos_ID")

    # seperating True and False then saving False
    TRUEonly = done_overlapped[done_overlapped["found_pos_ID"] == "True"]
    FALSEonly = done_overlapped[done_overlapped["found_pos_ID"] == "False"]
    print("Found posID TRUE df shape: ", TRUEonly.shape)
    print("Found posID FALSE df shape: ", FALSEonly.shape)

    # saving dfs
    FALSEonly.reset_index(drop=True,inplace=True)
    FALSEonly.to_csv(notfound_name,index=False)
    TRUEonly.reset_index(drop=True,inplace=True)
    TRUEonly.to_csv(found_name,index=False)

    return idset


In [84]:
# subsets filtered for 4093 detected IDs overlapping 11_2012 fasta
cpan = pd.read_csv("CYS_pantarget_11_2012IDs_6157.csv")
crea = pd.read_csv("CYS_reactivity_11_2012IDs_1483.csv")
kpan = pd.read_csv("LYS_pantarget_11_2012IDs_8350.csv")
krea = pd.read_csv("LYS_reactivity_11_2012IDs_4576.csv")

# 11_2012 release with 4093 detected IDs
#ukbfasta = pd.read_csv("sprot_11_2012_detectedIDs_4093.csv")
fastaC = pd.read_csv("sprot_11_2012_all_Cpos.csv")
fastaK = pd.read_csv("sprot_11_2012_all_Kpos.csv")
# make into list with only pos_ID
allC = list(fastaC.key_id)
allK = list(fastaK.key_id)
print()
CRidset = position_overlap(allC, crea, "pos_found_C_react.csv", "pos_not_found_C_react.csv")
print()
CPidset = position_overlap(allC, cpan, "pos_found_C_pan.csv", "pos_not_found_C_pan.csv")
print()
KRidset = position_overlap(allK, krea, "pos_found_K_react.csv", "pos_not_found_K_react.csv")
print()
KPidset = position_overlap(allK, kpan, "pos_found_K_pan.csv", "pos_not_found_K_pan.csv")
print()
totCid = set(list(CRidset) + list(CPidset))
print("total Cys group IDs: ", len(totCid))
totKid = set(list(KRidset) + list(KPidset))
print("total Lye group IDs: ", len(totKid))
allIDs = set(list(totCid) + list(totKid))
print("total IDs both C and K: ",len(allIDs))


chemo pos_ID length:  1483
overlapping AA's in chemoproteomic data subset:  1476
unique IDs in df:  1007
Found posID TRUE df shape:  (1476, 5)
Found posID FALSE df shape:  (7, 5)

chemo pos_ID length:  6157
overlapping AA's in chemoproteomic data subset:  6152
unique IDs in df:  2884
Found posID TRUE df shape:  (6152, 5)
Found posID FALSE df shape:  (5, 5)

chemo pos_ID length:  4576
overlapping AA's in chemoproteomic data subset:  4576
unique IDs in df:  1585
Found posID TRUE df shape:  (4576, 5)
Found posID FALSE df shape:  (0, 5)

chemo pos_ID length:  8350
overlapping AA's in chemoproteomic data subset:  8350
unique IDs in df:  2429
Found posID TRUE df shape:  (8350, 5)
Found posID FALSE df shape:  (0, 5)

total Cys group IDs:  3015
total Lye group IDs:  2691
total IDs both C and K:  4093


# stopping here

In [144]:
diffall = all2012 - all2018
print(len(diffall))

128


In [146]:
diffALL = pd.DataFrame(list(diffall))
diffALL.columns = ['notfoundIDs']
diffALL.head(10)

notfoundIDs
0      E7EVH7
1      Q8NDA2
2      E7ETI0
3      Q9H3P2
4      Q9Y5A7
5      Q5VZ89
6      Q9UKL0
7      P49750
8      H3BQ06
9      Q9Y4D8

In [148]:
# SAVING IDS NOT FOUND 
diffALL.to_csv("UKBID_notfound_allCK_128.csv", index=False)

In [61]:
# CYS IDS DIFFERENCE
diffC = everC2012 - c18
print(len(diffC))

diffK = everK2012 - k18
print(len(diffK))

101
57


In [63]:
diffC = pd.DataFrame(list(diffC))
diffC.columns = ['ID']

diffK = pd.DataFrame(list(diffK))
diffK.columns = ['ID']

diffC.head(3)

ID
0  E7EVH7
1  E7ETI0
2  Q9H3P2

In [64]:
diffK.head(3)

ID
0  Q8NDA2
1  Q9H3P2
2  Q9Y5A7

In [147]:
# saving IDS not found in 2018 data
diffC.to_csv("UKBID_notfound_everlabled_cys_101.csv",index=False)
diffK.to_csv("UKBID_notfound_everlabeled_lys_57.csv",index=False)

# [TODO 2] format Cys reactivity data by taking the average of values in column, then plot the reactivity of all Cys and Lys values to define what thresholds will be LOW-MEDIUM-HIGH cutoff

In [67]:
# turning series into list
listCreact = list(CreactTRUEonly.reactivity)

# turning list into list of lists
llcreact = [x.split() for x in listCreact]

In [69]:
llcreact[0:10]

[['20.00', '1/2/6'],
 ['5.74', '1/1/6'],
 ['20.00', '1/2/6'],
 ['0.90', '1/1/6'],
 ['1.86', '5.41', '2/4/6'],
 ['2.26', '1.88', '2.08', '2.29', '2.01', '2.12', '6/6/6'],
 ['5.79', '5.43', '2/2/6'],
 ['8.09', '1/1/6'],
 ['6.83', '3.16', '4.37', '3.99', '5.68', '5.81', '6/6/6'],
 ['7.56', '6.38', '2/2/6']]

In [70]:
for sublist in llcreact:
    del sublist[-1]
llcreact[0:10]
# removed last index of sublist which contained the / value

[['20.00'],
 ['5.74'],
 ['20.00'],
 ['0.90'],
 ['1.86', '5.41'],
 ['2.26', '1.88', '2.08', '2.29', '2.01', '2.12'],
 ['5.79', '5.43'],
 ['8.09'],
 ['6.83', '3.16', '4.37', '3.99', '5.68', '5.81'],
 ['7.56', '6.38']]

In [71]:
# getting average of lists in list
avgg = []
for sublist in llcreact:
    test_list = [float(i) for i in sublist]
    s = sum(test_list)
    l = len(sublist)
    a = s/l
    avgg.append(a)
avgg[0:10]

[20.0,
 5.74,
 20.0,
 0.9,
 3.6350000000000002,
 2.106666666666667,
 5.609999999999999,
 8.09,
 4.973333333333334,
 6.97]

In [72]:
# formating decimal place
avgf = ["{0:.2f}".format(x) for x in avgg]
avgf[0:10]

['20.00',
 '5.74',
 '20.00',
 '0.90',
 '3.64',
 '2.11',
 '5.61',
 '8.09',
 '4.97',
 '6.97']

In [73]:
print(len(avgf))

1492


In [74]:
# convert this into a column for CreactTRUE df
avgR = pd.DataFrame(np.array(avgf).reshape(-1,1))
avgR.columns = ['avgReactivity']
avgR.head(3)

avgReactivity
0         20.00
1          5.74
2         20.00

In [75]:
C2reactTRUEonly = pd.concat([CreactTRUEonly, avgR],axis=1)


In [76]:
Cfinalreact = C2reactTRUEonly[['pos_ID', 'avgReactivity']].copy()
Cfinalreact.head(3)

pos_ID avgReactivity
0   Q9Y6W6_C177         20.00
1    Q9Y6M9_C42          5.74
2  Q9Y6J0_C1197         20.00

In [77]:
Cfinalreact.columns = ['pos_ID','reactivity']

In [78]:
Kfinalreact = KreactTRUEonly[['pos_ID','reactivity']].copy()
Kfinalreact.head(3)

pos_ID  reactivity
0  P52758_K117         0.1
1   P02766_K35         0.1
2   P51572_K79         0.2

In [79]:
Kfinalreact.reactivity = pd.to_numeric(Kfinalreact.reactivity)
Cfinalreact.reactivity = pd.to_numeric(Cfinalreact.reactivity)

In [80]:
Cfinalreact = Cfinalreact.sort_values(by=['reactivity'])
Kfinalreact = Kfinalreact.sort_values(by=['reactivity'])

In [81]:
allreact = Kfinalreact.append(Cfinalreact,ignore_index=True)

In [82]:
allreact.reactivity = pd.to_numeric(allreact.reactivity)
allreact.dtypes

pos_ID         object
reactivity    float64
dtype: object

In [83]:
allreact = allreact.sort_values(by=['reactivity'])
allreact.reset_index(drop=True,inplace=True)

In [85]:
allreact.head(4)

pos_ID  reactivity
0  P52758_K117         0.1
1   P02766_K35         0.1
2   P51572_K79         0.2
3  P60900_K116         0.2

# [TODO 3] distribution of reactivity in LYS and CYS datasets

In [86]:
print(allreact.describe())

        reactivity
count  5996.000000
mean      7.427051
std       3.163972
min       0.100000
25%       5.000000
50%       8.700000
75%      10.000000
max      20.000000


In [87]:
# adding new threshold by old numbers from nature paper      
allreact.loc[allreact.reactivity <= 2, 'Threshold'] = 'High' 
allreact.loc[(allreact.reactivity > 2) & (allreact.reactivity <= 5), 'Threshold'] = 'Medium' 
allreact.loc[allreact.reactivity > 5, 'Threshold'] = 'Low'

from collections import Counter
thresholdf = Counter(list(allreact.Threshold))
thresdic = dict(thresholdf)
thresdic

{'High': 402, 'Medium': 1118, 'Low': 4476}

In [88]:
from collections import Counter
thresholdf = Counter(list(allreact.Threshold))
thresdic = dict(thresholdf)
thresdic

{'High': 402, 'Medium': 1118, 'Low': 4476}

# HIGH MED LOW COUNTS of all reactivity data

## {'High': 402, 'Medium': 1118, 'Low': 4476}

### HIGH : <= 2
### MEDIUM : 2+ & <= 5
### LOW : 5+

In [89]:
allreact[allreact["Threshold"] == "Low"].describe()

reactivity
count  4476.000000
mean      8.924229
std       1.995162
min       5.010000
25%       7.387500
50%      10.000000
75%      10.000000
max      20.000000

In [90]:
allreact[allreact["Threshold"] == "Medium"].describe()

reactivity
count  1118.000000
mean      3.649177
std       0.874127
min       2.010000
25%       2.900000
50%       3.700000
75%       4.460000
max       5.000000

In [91]:
allreact[allreact["Threshold"] == "High"].describe()

reactivity
count  402.000000
mean     1.263607
std      0.464303
min      0.100000
25%      0.900000
50%      1.300000
75%      1.692500
max      2.000000

# THRESHOLDs for CYS and LYS data seperately

In [92]:
Cfinalreact.reset_index(drop=True,inplace=True)
Kfinalreact.reset_index(drop=True,inplace=True)

In [93]:
Kfinalreact.head(10)

pos_ID  reactivity
0   P52758_K117         0.1
1    P02766_K35         0.1
2    P51572_K79         0.2
3   P60900_K116         0.2
4    Q86WQ0_K28         0.2
5    Q9NZ45_K68         0.2
6   P08237_K678         0.3
7    O14737_K66         0.3
8    P23284_K41         0.3
9  P53396_K1018         0.3

In [94]:
print(Kfinalreact.describe())

        reactivity
count  4504.000000
mean      7.926732
std       2.999950
min       0.100000
25%       5.800000
50%      10.000000
75%      10.000000
max      10.000000


In [95]:
Cfinalreact.head(10)

pos_ID  reactivity
0  O14733_C202        0.74
1   P07237_C56        0.80
2   P78417_C32        0.84
3  Q6DCA0_C153        0.88
4  Q9Y4X0_C175        0.88
5   Q00796_C45        0.89
6   Q9UHQ1_C99        0.89
7  Q9Y6I4_C302        0.90
8  P38606_C138        0.90
9   Q8IY18_C91        0.92

In [96]:
print(Cfinalreact.describe())

        reactivity
count  1492.000000
mean      5.918633
std       3.168556
min       0.740000
25%       4.290000
50%       5.560000
75%       6.830000
max      20.000000


In [97]:
# adding new threshold by old numbers from nature paper      
Cfinalreact.loc[Cfinalreact.reactivity <= 2, 'Threshold'] = 'High' 
Cfinalreact.loc[(Cfinalreact.reactivity > 2) & (Cfinalreact.reactivity <= 5), 'Threshold'] = 'Medium' 
Cfinalreact.loc[Cfinalreact.reactivity > 5, 'Threshold'] = 'Low'


In [98]:

from collections import Counter
Cthres = Counter(list(Cfinalreact.Threshold))
Cdic = dict(Cthres)
Cdic

{'High': 97, 'Medium': 451, 'Low': 944}

In [99]:
clow=Cfinalreact[Cfinalreact["Threshold"] == "Low"].describe()

clow.columns=['C_LOW']
clow

C_LOW
count  944.000000
mean     7.382680
std      3.027032
min      5.010000
25%      5.710000
50%      6.440000
75%      7.840000
max     20.000000

In [100]:
cmed = Cfinalreact[Cfinalreact["Threshold"] == "Medium"].describe()
cmed.columns=['C_MED']
cmed

C_MED
count  451.000000
mean     3.839645
std      0.827942
min      2.010000
25%      3.185000
50%      4.000000
75%      4.550000
max      5.000000

In [101]:
chigh=Cfinalreact[Cfinalreact["Threshold"] == "High"].describe()
chigh.columns=['C_HIGH']
chigh

C_HIGH
count  97.000000
mean    1.336804
std     0.340152
min     0.740000
25%     1.060000
50%     1.320000
75%     1.640000
max     1.960000

In [102]:
CLM = clow.join(cmed)
CLMH = CLM.join(chigh)
print(CLMH)


            C_LOW       C_MED     C_HIGH
count  944.000000  451.000000  97.000000
mean     7.382680    3.839645   1.336804
std      3.027032    0.827942   0.340152
min      5.010000    2.010000   0.740000
25%      5.710000    3.185000   1.060000
50%      6.440000    4.000000   1.320000
75%      7.840000    4.550000   1.640000
max     20.000000    5.000000   1.960000


In [103]:
# adding new threshold by old numbers from nature paper      
Kfinalreact.loc[Kfinalreact.reactivity <= 2, 'Threshold'] = 'High' 
Kfinalreact.loc[(Kfinalreact.reactivity > 2) & (Kfinalreact.reactivity <= 5), 'Threshold'] = 'Medium' 
Kfinalreact.loc[Kfinalreact.reactivity > 5, 'Threshold'] = 'Low'

In [104]:

from collections import Counter
kthre = Counter(list(Kfinalreact.Threshold))
kdic = dict(kthre)
kdic

{'High': 305, 'Medium': 667, 'Low': 3532}

In [105]:
klow = Kfinalreact[Kfinalreact["Threshold"] == "Low"].describe()
klow.columns = ['K_LOW']
klow

K_LOW
count  3532.000000
mean      9.336240
std       1.338915
min       5.100000
25%       9.900000
50%      10.000000
75%      10.000000
max      10.000000

In [106]:
kmed = Kfinalreact[Kfinalreact["Threshold"] == "Medium"].describe()
kmed.columns = ['K_MED']
kmed

K_MED
count  667.000000
mean     3.520390
std      0.881579
min      2.100000
25%      2.700000
50%      3.500000
75%      4.300000
max      5.000000

In [107]:
khigh=Kfinalreact[Kfinalreact["Threshold"] == "High"].describe()
khigh.columns=['K_HIGH']
khigh

K_HIGH
count  305.000000
mean     1.240328
std      0.495552
min      0.100000
25%      0.900000
50%      1.300000
75%      1.700000
max      2.000000

In [108]:
klm = klow.join(kmed)
klmh = klm.join(khigh)
print(klmh)

             K_LOW       K_MED      K_HIGH
count  3532.000000  667.000000  305.000000
mean      9.336240    3.520390    1.240328
std       1.338915    0.881579    0.495552
min       5.100000    2.100000    0.100000
25%       9.900000    2.700000    0.900000
50%      10.000000    3.500000    1.300000
75%      10.000000    4.300000    1.700000
max      10.000000    5.000000    2.000000


In [109]:
CpanTRUEonly.reset_index(drop=True,inplace=True)
CreactTRUEonly.reset_index(drop=True,inplace=True)

KpanTRUEonly.reset_index(drop=True,inplace=True)
KreactTRUEonly.reset_index(drop=True,inplace=True)

In [110]:
def checkColumnValues(df, col):
    print(df[col].value_counts().reset_index().rename(columns={'index':col, col:'Count'}))

In [111]:
checkColumnValues(CpanTRUEonly, "labelType")

       labelType  Count
0  Iodoacetamide   5254
1         Target    737


In [112]:
print(CpanTRUEonly.shape)
print(len(set(CpanTRUEonly.pos_ID)))

(5991, 4)
5991


In [113]:
checkColumnValues(KpanTRUEonly,"labelType")

  labelType  Count
0     IodoL   8102
1   TargetL    118


In [114]:
print(KpanTRUEonly.shape)
print(len(set(KpanTRUEonly.pos_ID)))

(8220, 4)
8220


In [115]:
Cpanlist = list(CpanTRUEonly.pos_ID)
Creaclist = list(CreactTRUEonly.pos_ID)

Kpanlist = list(KpanTRUEonly.pos_ID)
Kreaclist = list(KreactTRUEonly.pos_ID)

C = set(Cpanlist + Creaclist)
K = set(Kpanlist + Kreaclist)

print(len(C))
print(len(K))

6315
9162


In [116]:
# saving ever laveled columns
CC = list(C)
KK = list(K)

Cever = pd.DataFrame(np.array(CC).reshape(-1,1))
Cever.columns = ["Ever_labeled_Cys_set"]

Kever = pd.DataFrame(np.array(KK).reshape(-1,1))
Kever.columns = ["Ever_labeled_Lys_set"]



In [117]:
CpanTRUEonly.labelType.replace("Iodoacetamide","panReactive",inplace=True)


/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py:5890: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


In [118]:
checkColumnValues(CpanTRUEonly,"labelType")

     labelType  Count
0  panReactive   5254
1       Target    737


In [119]:
KpanTRUEonly.labelType.replace("IodoL","panReactive",inplace=True)
KpanTRUEonly.labelType.replace("TargetL","Target",inplace=True)

In [120]:
checkColumnValues(KpanTRUEonly,"labelType")

     labelType  Count
0  panReactive   8102
1       Target    118


In [121]:
Cfinalreact= Cfinalreact.assign(subset="Cysteine")
Cever = Cever.assign(subset="Cysteine")
CpanTRUEonly = CpanTRUEonly.assign(subset="Cysteine")

Kfinalreact = Kfinalreact.assign(subset="Lysine")
Kever = Kever.assign(subset="Lysine")
KpanTRUEonly = KpanTRUEonly.assign(subset="Lysine")

```python
csplit = Cever["Ever_labeled_Cys_set"].str.split("_",n=1,expand=True)
lsplit = Kever["Ever_labeled_Lys_set"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]
```

In [122]:
def splitUKBentry(df, dfcol):
    split = df[dfcol].str.split("_", n=1, expand=True)
    split.columns= ["entry","pos"]
    split2 = pd.concat([df, split], axis =1)
    return split2

In [123]:
Cever2 = splitUKBentry(Cever, "Ever_labeled_Cys_set")
Kever2 = splitUKBentry(Kever, "Ever_labeled_Lys_set")
Cfinalreact2 = splitUKBentry(Cfinalreact,"pos_ID")
Kfinalreact2 = splitUKBentry(Kfinalreact,"pos_ID")

CpanTRUEonly2 = splitUKBentry(CpanTRUEonly,"pos_ID")
KpanTRUEonly2 = splitUKBentry(KpanTRUEonly,"pos_ID")

In [126]:
Cever2.head(10)

Ever_labeled_Cys_set    subset   entry    pos
0         P12270_C1068  Cysteine  P12270  C1068
1         P41229_C1047  Cysteine  P41229  C1047
2          Q92835_C956  Cysteine  Q92835   C956
3          O95347_C326  Cysteine  O95347   C326
4          P27708_C484  Cysteine  P27708   C484
5          Q14566_C540  Cysteine  Q14566   C540
6          P61163_C222  Cysteine  P61163   C222
7           Q8TBC4_C82  Cysteine  Q8TBC4    C82
8          Q9H9A5_C633  Cysteine  Q9H9A5   C633
9          Q9BV79_C263  Cysteine  Q9BV79   C263

In [127]:
Kever2.head(10)

Ever_labeled_Lys_set  subset   entry   pos
0           P61081_K36  Lysine  P61081   K36
1          Q15437_K237  Lysine  Q15437  K237
2          P05198_K154  Lysine  P05198  K154
3          P38646_K175  Lysine  P38646  K175
4          P18621_K121  Lysine  P18621  K121
5          O95479_K208  Lysine  O95479  K208
6          P02545_K108  Lysine  P02545  K108
7          P04406_K145  Lysine  P04406  K145
8          P26641_K147  Lysine  P26641  K147
9          P31948_K366  Lysine  P31948  K366

In [128]:
Cfinalreact2.head(10)

pos_ID  reactivity Threshold    subset   entry   pos
0  O14733_C202        0.74      High  Cysteine  O14733  C202
1   P07237_C56        0.80      High  Cysteine  P07237   C56
2   P78417_C32        0.84      High  Cysteine  P78417   C32
3  Q6DCA0_C153        0.88      High  Cysteine  Q6DCA0  C153
4  Q9Y4X0_C175        0.88      High  Cysteine  Q9Y4X0  C175
5   Q00796_C45        0.89      High  Cysteine  Q00796   C45
6   Q9UHQ1_C99        0.89      High  Cysteine  Q9UHQ1   C99
7  Q9Y6I4_C302        0.90      High  Cysteine  Q9Y6I4  C302
8  P38606_C138        0.90      High  Cysteine  P38606  C138
9   Q8IY18_C91        0.92      High  Cysteine  Q8IY18   C91

In [129]:
Kfinalreact2.head(10)

pos_ID  reactivity Threshold  subset   entry    pos
0   P52758_K117         0.1      High  Lysine  P52758   K117
1    P02766_K35         0.1      High  Lysine  P02766    K35
2    P51572_K79         0.2      High  Lysine  P51572    K79
3   P60900_K116         0.2      High  Lysine  P60900   K116
4    Q86WQ0_K28         0.2      High  Lysine  Q86WQ0    K28
5    Q9NZ45_K68         0.2      High  Lysine  Q9NZ45    K68
6   P08237_K678         0.3      High  Lysine  P08237   K678
7    O14737_K66         0.3      High  Lysine  O14737    K66
8    P23284_K41         0.3      High  Lysine  P23284    K41
9  P53396_K1018         0.3      High  Lysine  P53396  K1018

In [130]:
CpanTRUEonly2.head(10)

pos_ID    labelType      ID 2018pos_ID    subset   entry   pos
0  P26641_C266  panReactive  P26641       True  Cysteine  P26641  C266
1  P07900_C374  panReactive  P07900       True  Cysteine  P07900  C374
2  P23528_C139  panReactive  P23528       True  Cysteine  P23528  C139
3  P68104_C411  panReactive  P68104       True  Cysteine  P68104  C411
4  P08238_C366  panReactive  P08238       True  Cysteine  P08238  C366
5  P60981_C163  panReactive  P60981       True  Cysteine  P60981  C163
6  P61978_C132  panReactive  P61978       True  Cysteine  P61978  C132
7  P60709_C217  panReactive  P60709       True  Cysteine  P60709  C217
8   Q15365_C54  panReactive  Q15365       True  Cysteine  Q15365   C54
9  P60709_C272  panReactive  P60709       True  Cysteine  P60709  C272

In [131]:
KpanTRUEonly2.head(10)

pos_ID    labelType      ID 2018pos_ID  subset   entry    pos
0  A0AVT1_K1014  panReactive  A0AVT1       True  Lysine  A0AVT1  K1014
1   A0AVT1_K544  panReactive  A0AVT1       True  Lysine  A0AVT1   K544
2    A0AVT1_K86  panReactive  A0AVT1       True  Lysine  A0AVT1    K86
3   A0AVT1_K963  panReactive  A0AVT1       True  Lysine  A0AVT1   K963
4   A0JNW5_K645  panReactive  A0JNW5       True  Lysine  A0JNW5   K645
5   A2RUC4_K194  panReactive  A2RUC4       True  Lysine  A2RUC4   K194
6   A4D1E9_K222  panReactive  A4D1E9       True  Lysine  A4D1E9   K222
7   A4D1F6_K542  panReactive  A4D1F6       True  Lysine  A4D1F6   K542
8  A5YKK6_K1948  panReactive  A5YKK6       True  Lysine  A5YKK6  K1948
9   A6NDG6_K302  panReactive  A6NDG6       True  Lysine  A6NDG6   K302

# SAVING ALL FILES!

In [112]:
Cever2.to_csv("Cys_ever_labeled_set_6315.csv",index=False)
Kever2.to_csv("Lys_ever_labeled_set_9162.csv",index=False)

Cfinalreact2.to_csv("Cys_reactive_LMHset_1492.csv",index=False)
Kfinalreact2.to_csv("Lys_reactive_LMHset_4504.csv",index=False)

CpanTRUEonly2.to_csv("Cys_panTar_5991.csv",index=False)
KpanTRUEonly2.to_csv("Lys_panTar_8220.csv",index=False)

---
---
---
---

In [135]:
CpanTRUEonly.drop(['2018pos_ID'],inplace=True, axis=1)

In [136]:
KpanTRUEonly.drop(['2018pos_ID'],inplace=True, axis=1)

In [133]:
Cfinalreact.head(3)

pos_ID  reactivity Threshold    subset
0  O14733_C202        0.74      High  Cysteine
1   P07237_C56        0.80      High  Cysteine
2   P78417_C32        0.84      High  Cysteine

In [134]:
Cever.head(3)

Ever_labeled_Cys_set    subset
0         P12270_C1068  Cysteine
1         P41229_C1047  Cysteine
2          Q92835_C956  Cysteine

In [137]:
CpanTRUEonly.head(3)

pos_ID    labelType      ID    subset
0  P26641_C266  panReactive  P26641  Cysteine
1  P07900_C374  panReactive  P07900  Cysteine
2  P23528_C139  panReactive  P23528  Cysteine

In [138]:
KpanTRUEonly.head(3)

pos_ID    labelType      ID  subset
0  A0AVT1_K1014  panReactive  A0AVT1  Lysine
1   A0AVT1_K544  panReactive  A0AVT1  Lysine
2    A0AVT1_K86  panReactive  A0AVT1  Lysine

In [139]:
Kever.head(3)

Ever_labeled_Lys_set  subset
0           P61081_K36  Lysine
1          Q15437_K237  Lysine
2          P05198_K154  Lysine

In [140]:
Kfinalreact.head(3)

pos_ID  reactivity Threshold  subset
0  P52758_K117         0.1      High  Lysine
1   P02766_K35         0.1      High  Lysine
2   P51572_K79         0.2      High  Lysine

# looking at nonRedundant ever labeled set, which genes have been labeled the most:


In [149]:
csplit = Cever["Ever_labeled_Cys_set"].str.split("_",n=1,expand=True)
lsplit = Kever["Ever_labeled_Lys_set"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]

In [150]:
allGenes = list(csplit.entry) + list(lsplit.entry)
allG = set(allGenes)
print(len(allG))

3991


In [151]:
csplit.head(3)

entry position
0  P12270    C1068
1  P41229    C1047
2  Q92835     C956

In [152]:
lsplit.head(3)

entry position
0  P61081      K36
1  Q15437     K237
2  P05198     K154

In [161]:
csplitCOUNTS = csplit.loc[:,'entry'].value_counts()
csplitCOUNTS

P49327    25
P21333    25
Q15149    22
Q7Z6Z7    20
P78527    20
Q14204    18
O75369    18
Q9Y490    17
O00410    15
P13639    15
P46013    14
Q92616    14
P35579    14
Q9NQW6    13
Q00610    12
P49792    12
Q12888    12
P51610    12
P22314    11
Q14315    11
P27708    11
P21980    11
P13489    11
P42704    11
P07814    11
P22102    10
O95071    10
Q14980    10
Q01813    10
Q71U36    10
          ..
O75528     1
P41743     1
O15541     1
Q06136     1
Q96CU9     1
Q9H361     1
Q8TCU6     1
P11172     1
O14519     1
Q12849     1
Q86UW9     1
Q92665     1
P24390     1
P26196     1
Q14012     1
O15530     1
O75334     1
P49753     1
Q9UEY8     1
O75928     1
Q9ULZ2     1
P23443     1
Q8N3F8     1
P08729     1
Q9NWV8     1
Q9UBQ5     1
O15260     1
Q9NXJ5     1
Q99543     1
Q9UHY7     1
Name: entry, Length: 2939, dtype: int64

In [162]:
csplitCOUNTS.to_csv("CYS_EverLabeled_gene_valueCounts_2939.csv")

In [164]:
ksplitCOUNTS = lsplit.loc[:,'entry'].value_counts()
ksplitCOUNTS

Q09666    105
P21333     50
O75369     47
P35579     42
Q15149     36
P00558     30
P07900     29
P13639     28
P10809     28
P31948     27
P02545     26
P07237     26
P68104     25
P08238     25
Q9Y490     25
O43707     24
P06733     24
P14618     24
P11142     23
Q14247     22
P50990     20
Q01518     19
P11586     19
P14625     19
P53396     19
Q01813     19
Q92945     19
P30101     18
Q9H4M9     18
P50454     18
         ... 
Q9BQA9      1
Q96JY6      1
Q9Y241      1
O75928      1
Q9H2K2      1
Q9Y680      1
P55084      1
Q9HB40      1
Q8N9N8      1
O75822      1
Q8N2C7      1
Q96NB2      1
Q969G5      1
P54709      1
Q8WVF1      1
Q9BQ52      1
Q06547      1
Q9GZZ9      1
Q16695      1
Q9H0F7      1
Q9UJF2      1
P99999      1
O43929      1
O15514      1
O60880      1
Q86UL3      1
Q96S55      1
Q92882      1
P18858      1
Q86TI2      1
Name: entry, Length: 2635, dtype: int64

In [165]:
ksplitCOUNTS.to_csv("LYS_EverLabeled_gene_valueCounts_2635.csv")

In [166]:
alleverlabeled = csplit.append(lsplit, ignore_index = True)
alleverlabeled

entry position
0      P12270    C1068
1      P41229    C1047
2      Q92835     C956
3      O95347     C326
4      P27708     C484
5      Q14566     C540
6      P61163     C222
7      Q8TBC4      C82
8      Q9H9A5     C633
9      Q9BV79     C263
10     Q86W50     C253
11     Q96G25      C31
12     P14618     C152
13     P15407      C97
14     Q06330     C313
15     P52701      C88
16     P49327     C496
17     O96019      C32
18     P50454     C156
19     Q6IBS0     C141
20     Q2TAL8     C529
21     Q15334     C697
22     Q9NQ88     C114
23     Q9NYL2     C150
24     Q15149    C3336
25     P04183      C66
26     P53618     C684
27     Q9NPD3      C97
28     Q9H0W8     C411
29     Q8N556     C157
...       ...      ...
15447  P21333     K936
15448  Q8N4C8    K1128
15449  P06493     K279
15450  O14579     K278
15451  P55769      K86
15452  P62701      K94
15453  O00193      K91
15454  O15143      K87
15455  P05783     K426
15456  P27144     K186
15457  O43390     K187
15458  P11021     K376
15459  Q93008    K2227
15460  Q96DI7      K18
15461  Q6YN16     K318
15462  Q09666    K2589
15463  Q13630     K282
15464  Q96DG6     K159
15465  Q9NVX2     K397
15466  P35579      K29
15467  Q9H074     K253
15468  Q12851     K638
15469  Q9NQP4     K116
15470  O75575      K24
15471  P15311     K193
15472  P62979     K143
15473  Q9Y5V0      K34
15474  O60716     K749
15475  P52888       K2
15476  P51148      K23

[15477 rows x 2 columns]

In [167]:
# count how many times each gene has EVER been labeled by C or K (all C and K positions are unique!)
togethergenecount = alleverlabeled.loc[:,'entry'].value_counts()
togethergenecount

Q09666    112
P21333     75
O75369     65
Q15149     58
P35579     56
P13639     43
Q9Y490     42
P07900     36
P49327     36
P00558     35
P14618     32
P07237     32
P31948     32
P08238     31
P10809     31
O43707     30
P02545     30
P11586     29
Q14204     29
P68104     29
Q01813     29
P11142     27
P06733     27
O75083     26
Q99832     25
Q00610     25
P30101     25
O00410     25
P50990     24
P22234     24
         ... 
Q9GZZ1      1
Q96G74      1
Q9ULJ3      1
Q02218      1
P17540      1
P20936      1
Q9UBZ4      1
Q14814      1
P14923      1
P11532      1
Q0PNE2      1
P08138      1
Q9H3R5      1
Q9UBT7      1
Q96EV8      1
O95470      1
P35711      1
Q9BUH6      1
Q12788      1
P84101      1
Q13043      1
Q86WS5      1
O00515      1
P86791      1
Q9Y478      1
Q16512      1
O00329      1
P42345      1
O75152      1
Q7Z478      1
Name: entry, Length: 3991, dtype: int64

In [168]:
togethergenecount.to_csv("CYS_LYS_EverLabeled_gene_valueCounts_3991.csv")

# enteries overlapping between Cys and Lys


In [153]:
c_entry = list(csplit['entry'])
k_entry = list(lsplit['entry'])
c_set = set(c_entry)
k_set = set(k_entry)
print(len(c_set))
print(len(k_set))

2939
2635


In [154]:
len(list(c_set & k_set))

1583

In [155]:
inter = c_set.intersection(k_set)
len(inter) # overlapping entries between cys and lys data

1583

In [169]:
KminusC = k_set - c_set
CminusK = c_set - k_set
print(len(KminusC))
print(len(CminusK))

print(1583 + 1052) # 2635 matched unique lys entry
print(1583 + 1356) # 2939 matched unique cys entry

1052
1356
2635
2939


# finding genes that have most labeleing, want to count # of LOW MED HIGH

In [171]:
csplit = Cfinalreact["pos_ID"].str.split("_",n=1,expand=True)
lsplit = Kfinalreact["pos_ID"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]
cthr = Cfinalreact['Threshold']
kthr = Kfinalreact['Threshold']

Cpy = pd.concat([csplit,cthr],axis=1)
Kpy = pd.concat([lsplit,kthr],axis=1)

In [172]:
Cgroups = Cpy.groupby(['entry','Threshold']).count()
Kgroups = Kpy.groupby(['entry','Threshold']).count()

In [236]:
Cgroups.to_csv("Cys_entry_Threshold_grouping_PymolStruc.csv")
Kgroups.to_csv("Lys_entry_Threshold_grouping_PymolStruc.csv")

In [173]:
Cgroups

position
entry  Threshold          
A0AVT1 Medium            2
A0MZ66 Low               1
A6NDG6 Medium            1
A6NDU8 Low               1
A6NED2 High              1
A6NHL2 Medium            3
A6NNF4 Low               1
A9UHW6 Medium            1
O00170 Medium            2
O00244 Medium            2
O00267 High              1
O00268 Medium            1
O00273 Low               2
O00299 Low               2
       Medium            2
O00303 Low               1
O00429 Low               1
       Medium            1
O00487 Medium            1
O00506 Low               2
O00515 Low               1
O00541 Medium            1
O00743 Medium            1
O00764 Low               1
O14545 Low               1
O14556 Low               1
O14579 Medium            1
O14733 High              1
       Medium            1
O14744 Low               1
...                    ...
Q9Y3A3 Low               1
Q9Y3C6 Low               1
Q9Y3D0 High              1
Q9Y3D2 Low               1
       Medium            1
Q9Y3E2 Medium            2
Q9Y3F4 Low               1
Q9Y478 Medium            1
Q9Y490 Low               1
Q9Y4P1 Low               1
       Medium            1
Q9Y4R8 Low               1
Q9Y4X0 High              1
Q9Y508 Low               1
Q9Y5B9 Medium            1
Q9Y5K6 Low               1
Q9Y5N6 Low               1
Q9Y5P6 Low               1
Q9Y5R8 Low               1
Q9Y5S2 High              1
Q9Y5S9 Low               1
Q9Y5Y2 Low               1
       Medium            3
Q9Y617 Low               2
Q9Y696 Medium            1
Q9Y6E0 Medium            1
Q9Y6I4 High              1
Q9Y6J0 Low               1
Q9Y6M9 Low               1
Q9Y6W6 Low               1

[1130 rows x 1 columns]

In [174]:
Kgroups

position
entry  Threshold          
A0AVT1 Medium            1
A1L0T0 Low               1
A6NDG6 Low               2
A9UHW6 Low               1
O00148 Low               3
O00151 Low               3
       Medium            2
O00154 Low               7
       Medium            1
O00159 Medium            1
O00170 Low               3
       Medium            1
O00193 Low               2
O00203 High              1
O00231 Low               3
       Medium            1
O00232 Low               3
       Medium            1
O00233 Low               2
O00244 Low               1
       Medium            1
O00264 Low               1
O00299 High              2
       Low               5
       Medium            3
O00303 Low               3
O00330 Low               1
O00410 High              1
       Low               5
       Medium            1
...                    ...
Q9Y490 Low               9
       Medium            2
Q9Y4L1 Low               9
       Medium            1
Q9Y4X0 High              1
Q9Y4Z0 High              1
       Low               1
Q9Y512 Low               1
Q9Y530 Low               1
Q9Y547 Low               1
Q9Y570 Low               2
Q9Y5J7 Low               1
Q9Y5K5 Medium            1
Q9Y5L0 High              1
       Low               1
Q9Y5L4 Low               1
Q9Y5Q0 Low               1
Q9Y5U9 Low               2
Q9Y5Y2 Low               1
Q9Y617 Low               4
       Medium            4
Q9Y673 Low               1
Q9Y678 High              1
Q9Y696 High              1
       Low               3
       Medium            3
Q9Y6A4 Medium            1
Q9Y6A5 High              1
Q9Y6C9 Medium            1
Q9Y6E2 Low               1

[2047 rows x 1 columns]